core/common/enum.hpp

```diff
@@ -5,9 +5,20 @@
 
 #pragma once
 
+#include <array>
+#include <string_view>
 #include <type_traits>
+#include <boost/optional.hpp>
 
 namespace fc::common {
+  template<typename Enumeration, size_t Number>
+  using ConversionTable = std::array<std::pair<Enumeration, std::string_view>, Number>;
```

Comment:
Could I suggest `std::map`?

Reply:
You also could use `std::map::find` to find by key and `std::find_if` with lambda to find by value.

Reply:
Maybe `array` was intentional, to avoid including `map` in header files, and because there are few elements.  
Also it would be two `map`s or `boost::bimap`, because we convert in both directions